src/repo.rs

```diff
@@ -811,6 +825,30 @@ impl Repository {
         }
     }
 
+    /// Create object database reading stream
+    pub fn odb_reader(&self, oid: Oid) -> Result<OdbReader, Error> {
+        let mut out = ptr::null_mut();
+        unsafe {
+            let mut db = ptr::null_mut();
+            try_call!(raw::git_repository_odb(&mut db, self.raw()));
+            try_call!(raw::git_odb_open_rstream(&mut out, db, oid.raw()));
+            raw::git_odb_free(db);
```

Comment:
I think this is leaked if the `open_rstream` fails, but in general it looks like there's room for an `Odb` abstraction here?

I'd imagine that, given enough time at least, there'd be an `odb` method and then from that `Odb` object you could create readers/writers?